algosdk/template.py

```diff
@@ -0,0 +1,280 @@
+    set ratn and ratd to 30 and 100, respectively. 
+    Split also have an expiry round, in which the owner can transfer back the assets.
+    ...
+
+    Arguments:
+        owner (str): an address that can receive the asset after the expiry round
+        receiver_1 (str): first address to receive assets 
+        receiver_2 (str): second address to receive assets 
+        ratn (int): the numerator of the first address fraction
+        ratd (int): the denominator of the first address fraction
+        expiry_round (int): the round on which the assets can be transferred back to owner 
+        min_pay (int): the minimum number of assets that can be transferred from the account
+        max_fee (int): half the maximum fee that can be paid to the network by the account
+
+    """
+    def __init__(self, owner : str, receiver_1: str, receiver_2: str, ratn : int, ratd : int, expiry_round : int, min_pay : int, max_fee : int):
+        self.owner = owner
+        self.receiver_1 = receiver_1
+        self.receiver_2 = receiver_2
+        self.ratn = ratn
+        self.ratd = ratd
+        self.expiry_round = expiry_round
+        self.min_pay = min_pay
+        self.max_fee = max_fee
+        
+    def get_program(self):
+        """
+        returns a byte array to be used in a 
+        """ 
+        orig = "ASAIAQUCAAYHCAkmAyDYHIR7TIW5eM/WAZcXdEDqv7BD+baMN6i2/A5JatGbNCDKsaoZHPQ3Zg8zZB/BZ1oDgt77LGo5np3rbto3/gloTyB40AS2H3I72YCbDk4hKpm7J7NnFy2Xrt39TJG0ORFg+zEQIhIxASMMEDIEJBJAABkxCSgSMQcyAxIQMQglEhAxAiEEDRAiQAAuMwAAMwEAEjEJMgMSEDMABykSEDMBByoSEDMACCEFCzMBCCEGCxIQMwAIIQcPEBA="
+        orig = base64.b64decode(orig)
+        output = base64.b64encode(inject(orig, [4, 7, 8, 9, 10, 14, 47, 80], [self.max_fee, self.expiry_round, self.ratn, self.ratd, self.min_pay, self.owner, self.receiver_1, self.receiver_2], [int, int, int, int, int, "address", "address", "address"])).decode()
+        return output
+        
+    def get_send_funds_transaction(self, first_valid_round, gh, amount : int, precise = True):
+        """
+        Return a group transactions array which transfer funds according to the contract's ratio.
+        
+        Parameters
+        ----------
+        amount : int
+            the amount to be transferred
+            
+        precise : bool, optional
+            precise treats the case where amount is not perfectly divisible based on the ration. 
+            When set to False, the amount will be divided as close as possible but one address will get 
+            slightly more. When True, an error will be raised. Defaults to True.
+
+        Raises
+        ------
+        NotDivisibleError
+            see precise
+        """
+        amt_1 = 0
+        amt_2 = 0
+        
+        gcd = math.gcd(self.ratn, self.ratd)
+        ratn = self.ratn // gcd
+        ratd = self.ratd // gcd
+
+        if amount % ratd == 0:
+            amt_1 = amount // ratd * ratn
+            amt_2 = amount - amt_2
```

Comment:
I know this is a draft but I spotted this while looking at how you were implementing `precise` logic - `amt_2 = amount - amt_1`

Reply:
Also I like how you're implementing `precise`, copying that for js-sdk